Change fork to newly structured repo for our needs

jebeck opened the pull request:
What it says on the tin.

jebeck committed:
array indexes in path represented as string

jebeck committed:
add a Travis config

jebeck committed:
add Travis build badge to README

leoasis committed:
Merge pull request #8 from tidepool-org/test-fix-and-travis

fix for broken tests?

GordyD committed:
Change fork to newly structured repo for our needs

GordyD committed:
Merge from fork

jebeck commented:
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;8F7bDivf&#x2F;17-redux-blip-uploader-publish-our-fork-of-redux-immutable-state-invariant-to-npm-as-a-object-invariant-test-helper"><img src="https:&#x2F;&#x2F;github.trello.services&#x2F;images&#x2F;trello-icon.png" width="12" height="12"> [redux] [blip] [uploader] publish our fork of `redux-immutable-state-invariant` to npm as a `object-invariant-test-helper`</a>

jebeck commented on .travis.yml:
```diff
@@ -0,0 +1,10 @@
+language: node_js
```
Oh want to add `sudo: false` at the top here? That allows faster builds on Travis I recently learned.

GordyD replied:
Sure

jebeck commented on README.md:
```diff
@@ -1,42 +1,20 @@
+A module that is useful for when needing to test for immutability.
 
-## Why?
+This module began life as a fork of redux-immutable-state-invariant by Leonardo Andres Garcia Crespo.
```
Put a link behind this credit to his GitHub profile? And a link behind the `redux-imm...` to the original project?

GordyD replied:
Ok!

jebeck commented on README.md:
```diff
@@ -1,42 +1,20 @@
-As said above, **don't use this in production!** It involves a lot of object copying and will degrade your app's performance. This is intended to be a tool to aid you in development and help you catch bugs.
+## Usage
 
-To use it, just add it as a middleware in your redux store:
```
Let's throw an example test in here.

GordyD replied:
I'm not sure what this looks like yet, haven't had a chance to test. Do you have a snippet?

jebeck commented on package.json:
```diff
@@ -1,24 +1,18 @@
+    "prepublish": "npm run clean && npm run build"
   },
-  "author": "Leonardo Andres Garcia Crespo <leoasis@gmail.com>",
+  "author": "Jana Beck <jana@tidepool.org>",
```
Hmm, I think npm has a ["contributors" alternative](https://docs.npmjs.com/files/package.json#people-fields-author-contributors), maybe we should use that to include original author? And add you if you like ;)

GordyD replied:
Yea, good idea, on it!

GordyD committed:
Updates following PR comments

Add `sudo: false` to travis.yml
Update readme to include link back to Leonardo's original repo which gave us inspiration
Update package.json to include contributors instead of single author

jebeck committed:
update node version to v0.12.7

jebeck committed:
misc polish

jebeck committed:
update exports and add example tests

GordyD commented on README.md:
```diff
@@ -1,42 +1,41 @@
+    });
+
+    expect(finalState).toEqual(expectedState);
+    expect(stateInvariantTestHelper.hasMutated(tracked)).toEqual(false);
```
Looks good. This API feels a lot cleaner to me.

jebeck committed:
- export the helper with default isImmutable by default to simplify interface
- (consistency in) naming is hard; update to objectInvariantTestHelper (like the package name) everywhere
